apps/agent/src/mastra/workflows/github-send-today-list-PRs-to-discord.ts

```diff
@@ -15,295 +10,656 @@ import { formatDate } from '../../utils/datetime'
+
+// Schema for channel data with repositories
+const channelDataSchema = z.object({
+  orgId: z.number(),
+  orgName: z.string(),
+  channelId: z.string(),
+  channelName: z.string(),
+  platform: z.enum(['discord', 'slack']),
+  repositories: z.array(repoDataSchema),
 })
 
-type StepThreeOutput = z.infer<typeof stepThreeSchema>
+type ChannelData = z.infer<typeof channelDataSchema>
 
-const stepThree = new Step({
-  id: 'compose-message',
-  outputSchema: stepThreeSchema,
+// Step 2: Fetch PRs and commits for each repository
+const stepFetchRepoData = new Step({
+  id: 'fetch-repo-data',
+  outputSchema: z.object({
+    channelDataList: z.array(channelDataSchema),
+  }),
   execute: async ({ context }) => {
-    if (
-      context.steps['get-today-pr-list']?.status !== 'success' ||
-      context.steps['get-today-commits']?.status !== 'success'
-    ) {
-      throw new Error('Failed to get today PR list or commits')
+    if (context.steps['fetch-organizations']?.status !== 'success') {
+      throw new Error('Failed to fetch organizations')
     }
 
-    const { todayPRs } = context.steps['get-today-pr-list']
-      ?.output as StepOneOutput
-
-    const { todayCommits } = context.steps['get-today-commits']
-      ?.output as StepTwoOutput
-
-    const openPRs = todayPRs.filter((pr) => !pr.isMerged && !pr.isWIP) || []
-    const mergedPRs = todayPRs.filter((pr) => pr.isMerged) || []
-    const wipPRs = todayPRs.filter((pr) => pr.isWIP) || []
-    const needToReviewPRs = todayPRs.filter(
-      (pr) => !pr.isMerged && !pr.isWIP && pr.isWaitingForReview,
-    )
-
-    const summary = [
-      { label: 'Open PRs', value: `**${openPRs.length}**` },
-      { label: 'Merged PRs', value: `**${mergedPRs.length}**` },
-      { label: 'WIP PRs', value: `**${wipPRs.length}**` },
-      { label: 'Commits', value: `**${todayCommits.length}**` },
-    ]
-
-    const representData = [
-      '🔥 **Summary**',
-      convertArrayToMarkdownTableList(summary, false),
-    ]
-
-    if (openPRs.length > 0) {
-      representData.push(
-        convertNestedArrayToTreeList({
-          label: '`Open PRs:`',
-          children: openPRs.map((pr) => ({
-            label: `[#${pr.number}](${pr.url}) ${pr.title}`,
-          })),
-        }),
-      )
-    }
+    const { organizations } = context.steps['fetch-organizations'].output
+    const channelDataList: ChannelData[] = []
 
-    if (mergedPRs.length > 0) {
-      representData.push(
-        convertNestedArrayToTreeList({
-          label: '\n`Merged PRs:`',
-          children: mergedPRs.map((pr) => ({
-            label: `[#${pr.number}](${pr.url}) ${pr.title}`,
-          })),
-        }),
-      )
-    }
+    for (const org of organizations) {
+      // Create GitHub client for this organization
+      const githubClient = new GitHubClient({
+        githubOwner: org.githubName,
+        githubToken: org.githubTokenId || process.env.GITHUB_TOKEN!, // Fallback to default token if not set
```

Comment:
```suggestion
        githubToken: org.githubTokenId,
```

GITHUB_TOKEN will be removed